### PR TITLE
ShareInternally: Make stale short link expiry optional

### PR DESCRIPTION
**Pull request by igorwwwwwwwwwwwwwwwwwwww**
**What is this feature?**

This expands on the work done by @gelicia in https://github.com/grafana/grafana/pull/86003. It allows stale short link expiry to be disabled by setting `short_links.expire_time` to a negative value.

**Why do we need this feature?**

A common pattern is to generate a short link and post it to an issue. This short link may not be accessed right away. Yet we will still want to retain it, in case someone comes looking for it later on.

For this reason, we want the option to disable the link expiry behaviour.

**Who is this feature for?**

Devs and SREs who use short links to link back to the sources of their analysis in issues and incident reviews.

**Which issue(s) does this PR fix?**:

- https://community.grafana.com/t/short-link-shows-404-after-some-time/74297
- Related: https://github.com/grafana/grafana/issues/83009

**Special notes for your reviewer:**

I could use some help actually testing this change, as I don't have a viable testing setup right now.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 1]
```diff
@@ -1785,7 +1785,7 @@ Configures settings around the short link feature.
 
 ### expire_time
 
-Short links which are never accessed are considered expired or stale, and will be deleted as cleanup. Set the expiration time in days. Default is `7` days. Maximum is `365` days, and setting above the maximum will have `365` set instead. Setting `0` means the short links will be cleaned up approximately every 10 minutes.
+Short links which are never accessed are considered expired or stale, and will be deleted as cleanup. Set the expiration time in days. Default is `7` days. Maximum is `365` days, and setting above the maximum will have `365` set instead. Setting `0` means the short links will be cleaned up approximately every 10 minutes. A negative value such as `-1` will disable expiry.
```
Comment (chri2547) [inline docs/sources/setup-grafana/configure-grafana/_index.md:1788]:
```suggestion
Short links that are never accessed are considered expired or stale and will be deleted as cleanup. Set the expiration time in days. The default is `7` days. The maximum is `365` days, and setting above the maximum will have `365` set instead. Setting `0` means the short links will be cleaned up approximately every 10 minutes. A negative value such as `-1` will disable expiry.
```

**Review by chri2547 — COMMENTED** [review]
Just a few suggested adjustments. Overall, the doc looks fine so I am approving. Thank you for the contribution!

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
@chri2547 I've applied the suggestion. Is anything else needed here?

**Comment by mtrin** [comment]
This one will be a nice to have :)
Thanks for the contribution

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Another bump to avoid this getting auto-closed.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
@chri2547 @mtrin Can we get this re-opened and maybe merged?

**Comment by chri2547** [comment]
@grafana/backend-enterprise @grafana/backend-platform - I'm not sure who should reopen and approve this PR. The docs are fine. Thanks.

**Comment by mtrin** [comment]
@igorwwwwwwwwwwwwwwwwwwww I don't actually work at Grafana, I believe @chri2547 or @torkelo may be able to reopen

**Review by irenerl24 — APPROVED** [review]
Did a minor commit in `defaults.ini` to fix style, other than that, LGTM! Thank you so much!

**Review by juanicabanas — APPROVED** [review]

**Comment by jdbaldry** [comment]
I've updated the branch so this looks good to go

**Pull request merged**